components/httpsrv/server/http_client.go

```diff
@@ -93,18 +93,18 @@ func NewHTTPClient(brk Broker) *HTTPClient {
 	}
 }
 
-func (c *HTTPClient) SendEvent(req *BrokerEventContext) error {
-	if req.TargetAdapter == nil {
-		return core.ErrInvalid(fmt.Errorf("adapter is missing"))
-	}
-	adapter, ok := req.TargetAdapter.(*v1alpha1.HTTPAdapter)
-	if !ok {
-		return core.ErrInvalid(fmt.Errorf("adapter is not HTTPAdapter"))
-	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), req.TTL())
+func (c *HTTPClient) SendEvent(req *grpc.ComponentEvent) error {
+	ctx, cancel := context.WithTimeout(context.Background(), req.Event.TTL())
 	log := c.log.WithEvent(req.Event)
 
+	adapter := &v1alpha1.HTTPAdapter{}
+	err := req.Event.Spec(&adapter.Spec)
+
+	if err != nil {
+		cancel()
+		return core.ErrInvalid(fmt.Errorf("error parsing adapter spec: %v", err))
+	}
+
```

Comment:
When only an error is returned it's more common to have the assignment within the if stmt.

```go
	if err := req.Event.Spec(&adapter.Spec); err != nil {
		cancel()
		return core.ErrInvalid(fmt.Errorf("error parsing adapter spec: %v", err))
	}
```

Reply:
Updated in https://github.com/xigxog/kubefox/pull/29/commits/cc9526c3a0e1fcfa278d5001efaa9dee2857fa72